New trainings for mvaTauID + one backport from CMSSW_10_0_X

roger-wolf opened the pull request:
Dear Slava, Andrea, 

this PR consists of a modification of the miniAOD sequences, adding new mvaTauID discriminants that have been trained with 2017 MCv1 (tauIdMVAIsoDBoldDMwLT2017). Modified files are:

   * RecoTauTag/Configuration/python/loadRecoTauTagMVAsFromPrepDB_cfi.py
   *  RecoTauTag/Configuration/python/updateHPSPFTaus.py
   *  PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py

This modification impacts both the standard and legacy80X workflows via an updated makePatTausTask.

In addition the protection to prevent the PATTauProducer from asserting when encountering requests for tauID WP that are not available on existing AOD, has been backported from 100X (*). Changed files are: 

   *  PhysicsTools/PatAlgos/plugins/PATTauProducer.cc
   *  PhysicsTools/PatAlgos/plugins/PATTauProducer.h
   *  PhysicsTools/PatAlgos/python/producersLayer1/tauProducer_cfi.py

The development has been made by @mbluj. 

The changes wrt to status quo are in the mvaTauID output only. Performance comparisons and more detailed comments can be checked from here (**)

Please consider this PR for reminiAODv2. 

Cheers,
Roger

(*)
https://github.com/cms-sw/cmssw/pull/21186

(**)
https://github.com/cms-tau-pog/cmssw/pull/71

mbluj committed:
Add new payloads for MVAIso 2017v1

(cherry picked from commit 75e3b6a6f3176d8c88816f786a4bf7778b129ec4)

mbluj committed:
bugfix

(cherry picked from commit f409fc94a07df206b20a51368b2a8c1e95812ddf)

mbluj committed:
Add protection against a missing tauId source(s)

(cherry picked from commit 48c07598cf15d3f0dfa14afff9ec17101c45a475)

mbluj committed:
use nullptr rather than 0 for a null pointer

(cherry picked from commit b2d6c676d449316f37083c99004b83d9a3955111)

mbluj committed:
Add protection againist missing (calo) tauId sources

(cherry picked from commit 4feb9c9ed9f2da6d0260b10448de96d79d0b0ef3)

mbluj committed:
add a switch allowing/disallowing skipping missing tauId products in the event

(cherry picked from commit 31cdc5eb71e453c51e7a5b798022e1552800608c)

mbluj committed:
code-checks

(cherry picked from commit cc795b6d98a6d353784e323df4b970aebce17abd)

roger-wolf committed:
changing LogWarning printout such that it is not raised for each tau but only at first occurence per instance

(cherry picked from commit 01b008a7d2661de02cf5e13d34402dc86c708d72)

mbluj committed:
Add function modyfing makePatTausTask to include MVAIso tau-Id 2017v1 for reMiniAOD with 94X (X>3)

mbluj committed:
fix input parameters

mbluj committed:
Add MVAIso tau-Id with 2017v1 training to miniAOD sequences

roger-wolf committed:
Merge pull request #71 from mbluj/CMSSW_9_4_X_tauIdForMiniAODv2

Modified MVAIso tau-id 2017v1 for miniAODv2 with 94X (X>3)

cmsbuild commented:
A new Pull Request was created by  @roger-wolf (Roger Wolf) for CMSSW_9_4_X.

It involves the following packages:

PhysicsTools/PatAlgos
RecoTauTag/Configuration


@perrotta, @cmsbuild, @monttj, @slava77 can you please review it and eventually sign? Thanks.
@TaiSakuma, @gouskos, @imarches, @ahinzmann, @acaudron, @mmarionncern, @rappoccio, @jdamgov, @jdolen, @nhanvtran, @gpetruc, @gkasieczka, @schoef, @ferencek, @mverzett, @mariadalfonso, @pvmulder, @seemasharmafnal, @JyothsnaKomaragiri this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

slava77 commented:
@roger-wolf 
is everything introduced in this PR already available in the master branch?

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

slava77 commented on RecoTauTag/Configuration/python/updateHPSPFTaus.py:
```diff
@@ -0,0 +1,53 @@
+import FWCore.ParameterSet.Config as cms
```
the name of this file requires some disambiguation.
There is also updateHPSPFTaus_cff with completely different content.

perhaps customiseHPSPFTausForPAT  ?

roger-wolf replied:
I've followed your proposals. 
Cheers,
Roger

slava77 commented on RecoTauTag/Configuration/python/updateHPSPFTaus.py:
```diff
@@ -0,0 +1,53 @@
+    process.hpsPFTauDiscriminationByVVTightIsolationMVArun2v1DBoldDMwLTMVAIsoFor94XMiniv2.mapping[0].cut = cms.string("RecoTauTag_tauIdMVAIsoDBoldDMwLT2017v1_WPEff40")    
+    #create a new task and put the moduled therin
+    process.hpsPFTauMVAIsoFor94XMiniv2Task = cms.Task()
+    process.hpsPFTauMVAIsoFor94XMiniv2Task.add(process.hpsPFTauDiscriminationByIsolationMVArun2v1DBoldDMwLTrawMVAIsoFor94XMiniv2)
```
it would be less verbose to simply pass all taggers in the Task constructor arguments.

roger-wolf commented:
Hi Slava, 

yes, everything that is commited here is also part of the master branch, with the exception of the 94X legacy processing specific parts of course. For more info I refer to @mbluj . I'll up date according to your comments tomorrow, as soon as I get to it. 

Cheers,
Roger

slava77 commented:
After some thought I think that we should make a commitment to support miniAOD processing using inputs from 94X Fall17 RECO/AOD production campaign.
So, this means an introduction of a modifier similar to run2_miniAOD_80XLegacy.
I propose to call it run2_miniAOD_94XFall17.
Since tau is currently the only customer, Roger, may I ask that you create this modifier.
Given this modifier we can then follow this:



- in master:
    - create run2_miniAOD_94XFall17
    - define hpsPFTauIsolationMVArun2v1DBoldDMwLTTask in HPSPFTaus_cff.py with the 8 needed modules 
        - use this task in HPSPFTaus_cff.py hpsPFTauMVAIsolation2Task
    - append hpsPFTauMVAIsoTask to makePatTausTask  in miniAOD_tools via run2_miniAOD_94XFall17
    - .. process.patTaus.tauIDSources remains unchanged
    - .. modules entering hpsPFTauMVAIsoTask remain unchanged
- in 94X (almost the same, but with minor modifications)
    - create run2_miniAOD_94XFall17
    - define hpsPFTauIsolationMVArun2v1DBoldDMwLTTask in in HPSPFTaus_cff.py with 7 needed modules 
        - add hpsPFTauDiscriminationByVVLooseIsolationMVArun2v1DBoldDMwLT only for this era
        - use this task in HPSPFTaus_cff.py hpsPFTauMVAIsolation2Task
    - append hpsPFTauMVAIsoTask to makePatTausTask  in miniAOD_tools via run2_miniAOD_94XFall17
    - process.patTaus.tauIDSources remains unchanged
    - modules entering hpsPFTauMVAIsoTask are modified via run2_miniAOD_94XFall17 to pick up the right payload values

let me know if this is clear enough or if we should discuss more.

cmsbuild commented:
-1

Tested at: 3f87478f1c8d5a5b52e78212a7e35fed221d086d


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/3f87478f1c8d5a5b52e78212a7e35fed221d086d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25653/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25653/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25653/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
4.22 step2
<pre>runTheMatrix-results/4.22_RunCosmics2011A+RunCosmics2011A+RECOCOSD+ALCACOSD+SKIMCOSD+HARVESTDC/step2_RunCosmics2011A+RunCosmics2011A+RECOCOSD+ALCACOSD+SKIMCOSD+HARVESTDC.log

</pre>136.7611 step2
<pre>runTheMatrix-results/136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM/step2_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM.log

</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/3f87478f1c8d5a5b52e78212a7e35fed221d086d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25653/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25653/git-merge-result

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

slava77 commented:
On 1/26/18 3:25 PM, cmsbuild wrote:

> *Failed tests: RelVals*
> 
>   * *RelVals*:
> 
> When I ran the RelVals I found an error in the following worklfows:
> 4.22 step2
> 
> 
> 136.7611 step2
> 


both errors are due to file I/O, not related to this PR

slava77 commented:
@cmsbuild please test

it looks like the failures are gone based on the recent jenkins tests

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 3f87478f1c8d5a5b52e78212a7e35fed221d086d
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25698/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/3f87478f1c8d5a5b52e78212a7e35fed221d086d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25698/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25698/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25698/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /build/cmsbld/jenkins/workspace/compare-root-files-short-matrix/results/JR-comparison/PR-21977/1325.7_TTbar_13_94XNanoAODINPUT+TTbar_13_94XNanoAODINPUT+NANOEDMMC2017

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 25
 - DQMHistoTests: Total histograms compared: 2717599
 - DQMHistoTests: Total failures: 105
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2717332
 - DQMHistoTests: Total skipped: 162
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0 KiB( 0 files compared)
 - Checked 106 log files, 9 edm output root files, 25 DQM output files

roger-wolf commented:
Hi Slava, 

FYI, I did not get to implement your changes from home, during the w/e. I'll try to do it in office, tomorrow before noon.

Cheers,
Roger

slava77 commented:
On 1/28/18 4:14 PM, Roger Wolf wrote:
> Hi Slava,
> 
> FYI, I did not get to implement your changes from home, during the w/e.
> I'll try to do it in office, tomorrow before noon.
> 

that's OK

The Jan 29 deadline was pushed back.
So, we have a few days to converge.

roger-wolf committed:
renaming python script for updating tauID to be used e.g. for the re-miniAOD campaign

roger-wolf committed:
renaming python script for updating tauID to be used e.g. for the re-miniAOD campaign

roger-wolf committed:
replacing add statements by constructor arguments as requested by reco

cmsbuild commented:
Pull request #21977 was updated. @perrotta, @cmsbuild, @monttj, @slava77 can you please check and sign again.

perrotta commented:
please test

mbluj commented:
On Mon, Jan 29, 2018 at 12:08 PM, Roger Wolf <notifications@github.com>
wrote:

> *@roger-wolf* commented on this pull request.
> ------------------------------
>
> In RecoTauTag/Configuration/python/updateHPSPFTaus.py
> <https://github.com/cms-sw/cmssw/pull/21977#discussion_r164399437>:
>
> > @@ -0,0 +1,53 @@
> +import FWCore.ParameterSet.Config as cms
>
> I've followed your proposals.
>
You should also correct this line:
https://github.com/cms-sw/cmssw/pull/21977/files#diff-76dede28e7751d7bb6c5c9542de4155cR327

M.

> Cheers,
> Roger
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/21977#discussion_r164399437>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AE-odd-mh29slpwcY22G4ashBE2APTmTks5tPabAgaJpZM4RulK1>
> .
>



-- 
Michał Bluj
National Centre for Nuclear Research / Narodowe Centrum Badań Jądrowych
Hoża 69
00-681 Warszawa, Poland
tel. (+48) 22 55 31 833
e-mail: michal.bluj@ncbj.gov.pl

cmsbuild commented:
The tests are being triggered in jenkins.

perrotta commented:
abort test

cmsbuild commented:
Jenkins tests are aborted.

roger-wolf committed:
and updating the link to re-miniAOD...

roger-wolf commented:
Thanx for the reminder Michal.

perrotta commented:
please test

slava77 commented:
@roger-wolf 
I noticed that the updates requested in https://github.com/cms-sw/cmssw/pull/21977#issuecomment-360923538
were not made.
Please clarify on the plans to implement them
Thank you.

roger-wolf commented:
Hi Slava,

I'm fine to do this. I need to understand first better how these 
modifiers work. Slava can you point me to the one you are mentioning 
with a few explanations and a running example? Otherwise I'm sure @mbluj 
can teach me a bit.

I'd try to do it asap this week an let you know if needed more time. 
I'll bug you withe questions if needed.

Cheers,
Roger



Am 26.01.2018 um 23:35 schrieb Slava Krutelyov:
>
> After some thought I think that we should make a commitment to support 
> miniAOD processing using inputs from 94X Fall17 RECO/AOD production 
> campaign.
> So, this means an introduction of a modifier similar to 
> run2_miniAOD_80XLegacy.
> I propose to call it run2_miniAOD_94XFall17.
> Since tau is currently the only customer, Roger, may I ask that you 
> create this modifier.
> Given this modifier we can then follow this:
>
>   * in master:
>       o create run2_miniAOD_94XFall17
>       o define hpsPFTauIsolationMVArun2v1DBoldDMwLTTask in
>         HPSPFTaus_cff.py with the 8 needed modules
>           + use this task in HPSPFTaus_cff.py hpsPFTauMVAIsolation2Task
>       o append hpsPFTauMVAIsoTask to makePatTausTask in miniAOD_tools
>         via run2_miniAOD_94XFall17
>       o .. process.patTaus.tauIDSources remains unchanged
>       o .. modules entering hpsPFTauMVAIsoTask remain unchanged
>   * in 94X (almost the same, but with minor modifications)
>       o create run2_miniAOD_94XFall17
>       o define hpsPFTauIsolationMVArun2v1DBoldDMwLTTask in in
>         HPSPFTaus_cff.py with 7 needed modules
>           + add
>             hpsPFTauDiscriminationByVVLooseIsolationMVArun2v1DBoldDMwLT
>             only for this era
>           + use this task in HPSPFTaus_cff.py hpsPFTauMVAIsolation2Task
>       o append hpsPFTauMVAIsoTask to makePatTausTask in miniAOD_tools
>         via run2_miniAOD_94XFall17
>       o process.patTaus.tauIDSources remains unchanged
>       o modules entering hpsPFTauMVAIsoTask are modified via
>         run2_miniAOD_94XFall17 to pick up the right payload values
>
> let me know if this is clear enough or if we should discuss more.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/21977#issuecomment-360923538>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AEwCUWg-21dagsdnZz6vLqEwYV9bXPMFks5tOlMmgaJpZM4RulK1>.
>

cmsbuild commented:
+1
Tested at: 3ab9cbb97cf21902b9f119df54083769a4f0a806
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25730/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/3f87478f1c8d5a5b52e78212a7e35fed221d086d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25730/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25730/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25730/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /build/cmsbld/jenkins/workspace/compare-root-files-short-matrix/results/JR-comparison/PR-21977/1325.7_TTbar_13_94XNanoAODINPUT+TTbar_13_94XNanoAODINPUT+NANOEDMMC2017

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 25
 - DQMHistoTests: Total histograms compared: 2717599
 - DQMHistoTests: Total failures: 106
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2717331
 - DQMHistoTests: Total skipped: 162
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0 KiB( 0 files compared)
 - Checked 106 log files, 9 edm output root files, 25 DQM output files

slava77 commented:
On 1/29/18 7:08 AM, Roger Wolf wrote:
> Hi Slava,
> 
> I'm fine to do this. I need to understand first better how these
> modifiers work. Slava can you point me to the one you are mentioning
> with a few explanations and a running example? Otherwise I'm sure @mbluj
> can teach me a bit.
> 
> I'd try to do it asap this week an let you know if needed more time.
> I'll bug you withe questions if needed.
> 

Hi Roger,

The example for run2_miniAOD_94XFall17 is in
run2_miniAOD_80XLegacy, which aims to achieve the same target (to modify
the standard setup in order to be able to process 80X inputs with the
current release)

You can do a git grep run2_miniAOD_80XLegacy to see where it's created
and how it is used.
To enable it in the cmsDriver command arguments you will need to specify
this modifier in the --era
--era Run2_2017,run2_miniAOD_94XFall17

Hope this clarifies.

		--slava


> Cheers,
> Roger
> 
> 
> 
> Am 26.01.2018 um 23:35 schrieb Slava Krutelyov:
>>
>> After some thought I think that we should make a commitment to support
>> miniAOD processing using inputs from 94X Fall17 RECO/AOD production
>> campaign.
>> So, this means an introduction of a modifier similar to
>> run2_miniAOD_80XLegacy.
>> I propose to call it run2_miniAOD_94XFall17.
>> Since tau is currently the only customer, Roger, may I ask that you
>> create this modifier.
>> Given this modifier we can then follow this:
>>
>> * in master:
>> o create run2_miniAOD_94XFall17
>> o define hpsPFTauIsolationMVArun2v1DBoldDMwLTTask in
>> HPSPFTaus_cff.py with the 8 needed modules
>> + use this task in HPSPFTaus_cff.py hpsPFTauMVAIsolation2Task
>> o append hpsPFTauMVAIsoTask to makePatTausTask in miniAOD_tools
>> via run2_miniAOD_94XFall17
>> o .. process.patTaus.tauIDSources remains unchanged
>> o .. modules entering hpsPFTauMVAIsoTask remain unchanged
>> * in 94X (almost the same, but with minor modifications)
>> o create run2_miniAOD_94XFall17
>> o define hpsPFTauIsolationMVArun2v1DBoldDMwLTTask in in
>> HPSPFTaus_cff.py with 7 needed modules
>> + add
>> hpsPFTauDiscriminationByVVLooseIsolationMVArun2v1DBoldDMwLT
>> only for this era
>> + use this task in HPSPFTaus_cff.py hpsPFTauMVAIsolation2Task
>> o append hpsPFTauMVAIsoTask to makePatTausTask in miniAOD_tools
>> via run2_miniAOD_94XFall17
>> o process.patTaus.tauIDSources remains unchanged
>> o modules entering hpsPFTauMVAIsoTask are modified via
>> run2_miniAOD_94XFall17 to pick up the right payload values
>>
>> let me know if this is clear enough or if we should discuss more.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/cms-sw/cmssw/pull/21977#issuecomment-360923538>,
>> or mute the thread
>>
> <https://github.com/notifications/unsubscribe-auth/AEwCUWg-21dagsdnZz6vLqEwYV9bXPMFks5tOlMmgaJpZM4RulK1>.
>>
>

roger-wolf committed:
splitting hpsPFTauMVAIsolation2Task in sub-tasks

roger-wolf committed:
defining new era run2_miniAOD_94XFall17

cmsbuild commented:
Pull request #21977 was updated. @perrotta, @monttj, @cmsbuild, @franzoni, @slava77, @fabiocos, @davidlange6 can you please check and sign again.

roger-wolf committed:
this should have become obsolete now

cmsbuild commented:
Pull request #21977 was updated. @perrotta, @monttj, @cmsbuild, @franzoni, @slava77, @fabiocos, @davidlange6 can you please check and sign again.

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 99d14f70ca481df20164ae35b5385add9a7a250e
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25895/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/3f87478f1c8d5a5b52e78212a7e35fed221d086d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25895/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25895/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25895/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /build/cmsbld/jenkins/workspace/compare-root-files-short-matrix/results/JR-comparison/PR-21977/1325.7_TTbar_13_94XNanoAODINPUT+TTbar_13_94XNanoAODINPUT+NANOEDMMC2017

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 25
 - DQMHistoTests: Total histograms compared: 2718773
 - DQMHistoTests: Total failures: 104
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2718507
 - DQMHistoTests: Total skipped: 162
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0 KiB( 0 files compared)
 - Checked 106 log files, 9 edm output root files, 25 DQM output files

roger-wolf committed:
now also the proper payloads are picked up when running with era run2_miniAOD_94XFall17

cmsbuild commented:
Pull request #21977 was updated. @perrotta, @monttj, @cmsbuild, @franzoni, @slava77, @fabiocos, @davidlange6 can you please check and sign again.

roger-wolf commented:
This should be the final commit. Please note that @mbluj still wanted to make some final tests using this commit and post the results to this thread.

mbluj commented:
On Tue, Feb 6, 2018 at 4:02 PM, Roger Wolf <notifications@github.com> wrote:

> This should be the final commit. Please note that @mbluj
> <https://github.com/mbluj> still wanted to make some final tests using
> this commit and post the results to this thread.
>
While my tests are ongoing, I have already spoted one issue: a new VVLoose
discriminant is computed by not added to patTaus.

mbluj commented:
Other comment: I think that new MVA2017v1 based tau-Ids can be part of the `run2_miniAOD_80XLegacy` era by default rather than require a combination of the `run2_miniAOD_80XLegacy` and `run2_miniAOD_94XFall17` eras. In my opinion the current solution with combined eras is not mandatory as there is nothing like a default 80XLegacy so not need backward comparability, while it is potentially error prone.

slava77 commented:
On 2/6/18 8:35 AM, mbluj wrote:
> Other comment: I think that new MVA2017v1 based tau-Ids can be part of
> the |run2_miniAOD_80XLegacy| era by default rather than require a
> combination of the |run2_miniAOD_80XLegacy| and |run2_miniAOD_94XFall17|
> eras. In my opinion the current solution with combined eras is not
> mandatory as there is nothing like a default 80XLegacy so not need
> backward comparability, while it is potentially error prone.

Good point.

run2_miniAOD_80XLegacy can not be assumed to be combined with
run2_miniAOD_94XFall17
The use cases should be exclusive.

It sounds like the PR needs an update.

> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/21977#issuecomment-363481605>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbjbn9DSbug6z1mLIGv8FG0Ke2PGyks5tSH9RgaJpZM4RulK1>.
>

roger-wolf committed:
commented line on request by TauPOG

roger-wolf committed:
consistent switch of tauID MVA training and extention of tauID WPs by VVLoose

cmsbuild commented:
Pull request #21977 was updated. @perrotta, @monttj, @cmsbuild, @franzoni, @slava77, @fabiocos, @davidlange6 can you please check and sign again.

mbluj committed:
Add VVLoose WP patTaus in more economical and safe way

cmsbuild commented:
Pull request #21977 was updated. @perrotta, @monttj, @cmsbuild, @franzoni, @slava77, @fabiocos, @davidlange6 can you please check and sign again.

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 7ac37bb5abc5cf6e6a363aed6a665e620592d289
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25987/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/3f87478f1c8d5a5b52e78212a7e35fed221d086d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25987/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25987/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/25987/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /build/cmsbld/jenkins/workspace/compare-root-files-short-matrix/results/JR-comparison/PR-21977/1325.7_TTbar_13_94XNanoAODINPUT+TTbar_13_94XNanoAODINPUT+NANOEDMMC2017+HARVESTNANOAODMC2017

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 23 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2718773
 - DQMHistoTests: Total failures: 103
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2718508
 - DQMHistoTests: Total skipped: 162
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0 KiB( 0 files compared)
 - Checked 107 log files, 9 edm output root files, 26 DQM output files

mbluj commented:
Hello all,
is the current implementation of the PR fulfilling all your requirements or there is still something to be done by us? What is a timeline for a 94X release for MiniAODv2 of 2017 data?
From our side the performance of the PR is fine, i.e. all MVA-based tau-Ids are working as expected.

slava77 commented:
> is the current implementation of the PR fulfilling all your requirements

the review of the PR in the master branch goes first.

slava77 commented on PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py:
```diff
@@ -318,22 +318,161 @@ def miniAOD_customizeCommon(process):
+    _makePatTausTaskWithRetrainedMVATauID.add(process.hpsPFTauDiscriminationByVVLooseIsolationMVArun2v1DBoldDMwLT)
+    ## modify raw mva values
+    run2_miniAOD_94XFall17.toModify(process.hpsPFTauDiscriminationByIsolationMVArun2v1DBoldDMwLTraw, mvaName = "RecoTauTag_tauIdMVAIsoDBoldDMwLT2017v1")
+    run2_miniAOD_94XFall17.toModify(process.hpsPFTauDiscriminationByIsolationMVArun2v1DBoldDMwLTraw, mvaOpt  = "DBoldDMwLTwGJ")
```
all existing modules should be modified in the same file where they were defined.
If I'm not mistaken, it should be happening in RecoTauTag/Configuration/python/HPSPFTaus_cff.py
This will also allow for a more transparent application of the modifier for run2_miniAOD_80XLegacy and run2_miniAOD_94XFall17, e.g. with a syntax like
```python
for e in [ run2_miniAOD_80XLegacy, run2_miniAOD_94XFall17]:
e.toModify(hpsPFTauDiscriminationByIsolationMVArun2v1DBoldDMwLTraw, 
                   mvaName = "RecoTauTag_tauIdMVAIsoDBoldDMwLT2017v1", 
                   mvaOpt  = "DBoldDMwLTwGJ")
```
Readability can also be improved by not repeating toModify with the module name for every single variable.

At a glance, it seems like exactly the same modifications are applied in both cases.
For this, it may even be more straightforward to define one modifier, e.g. ```run2_tauLT2017v1_mva```, which can be added into both run2_miniAOD_80XLegacy and run2_miniAOD_94XFall17.
I'd leave this for some later PR in the master branch.

mbluj replied:
Dear Slava,
I'm afraid that moving modification of tau-Ids from `miniAOD_tools.py` to the file where they are defined (`RecoTauTag/Configuration/python/HPSPFTaus_cff.py`) will change the RECO/AOD event content when RECO sequences are run with the `run2_miniAOD_94XFall17` era flag. It is what we want to avoid, no?

roger-wolf replied:
@slava77 I'm also not sure about the quations by @mbluj . Slava, once you clarify I'm fine to upload the change that I've already made locally. I think the change requested for tauProducer_cfi.py is safe in an case.

mbluj replied:
OK, but there is not protection to use this era with cmsDriver for RECO. However, I agree that it will be misuse, and that it is generally a good practice to have a modifier where modified modules are defined. Finally, as far as I know there is not many users of taus from RECOAOD nowadays (apart validation).

slava77 replied:
Technically, there is no incompatibility issue even for RECO.

The "no change" restriction extends to existing production eras.
The production era for e.g. 2017 is Run2_2017 (alone).

slava77 commented on PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py:
```diff
@@ -318,22 +318,161 @@ def miniAOD_customizeCommon(process):
+    ## add VVLoose WP to PAT tau
+    _extTauIDSources = process.patTaus.tauIDSources.clone()
+    _extTauIDSources.byVVLooseIsolationMVArun2v1DBoldDMwLT = cms.InputTag("hpsPFTauDiscriminationByVVLooseIsolationMVArun2v1DBoldDMwLT")
+    run2_miniAOD_94XFall17.toModify(process.patTaus,
```
similarly, modifications should be done where the module is defined, 
in PhysicsTools/PatAlgos/python/producersLayer1/tauProducer_cfi.py

roger-wolf committed:
cleanup of era replacements

cmsbuild commented:
Pull request #21977 was updated. @perrotta, @monttj, @cmsbuild, @franzoni, @slava77, @fabiocos, @davidlange6 can you please check and sign again.

slava77 commented on Configuration/StandardSequences/python/Eras.py:
```diff
@@ -44,7 +44,7 @@ def __init__(self):
                            'trackingLowPU', 'trackingPhase1', 'trackingPhase1QuadProp', 'ctpps_2016', 'trackingPhase2PU140',
                            'tracker_apv_vfp30_2016', 'run2_miniAOD_80XLegacy', 'run2_nanoAOD_92X',
-                           'hcalHardcodeConditions', 'hcalSkipPacker']
+                           'hcalHardcodeConditions', 'hcalSkipPacker', 'run2_miniAOD_94XFall17']
```
it would be nice to have this edit in the same place as where it shows up in the master branch

slava77 commented on PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py:
```diff
@@ -318,22 +318,31 @@ def miniAOD_customizeCommon(process):
+    _makePatTausTaskWithRetrainedMVATauID.add(process.hpsPFTauDiscriminationByIsolationMVArun2v1DBoldDMwLTTask)
+    process.hpsPFTauDiscriminationByVVLooseIsolationMVArun2v1DBoldDMwLT = process.hpsPFTauDiscriminationByVLooseIsolationMVArun2v1DBoldDMwLT.clone()
+    process.hpsPFTauDiscriminationByVVLooseIsolationMVArun2v1DBoldDMwLT.mapping[0].cut = cms.string("RecoTauTag_tauIdMVAIsoDBoldDMwLT2017v1_WPEff95")
+    _makePatTausTaskWithRetrainedMVATauID.add(process.hpsPFTauDiscriminationByVVLooseIsolationMVArun2v1DBoldDMwLT)
```
please add this module to hpsPFTauDiscriminationByIsolationMVArun2v1DBoldDMwLTTask in RecoTauTag/Configuration/python/HPSPFTaus_cff.py
This will  minimize required knowledge of the tau reco in the miniAOD_tools.
(this was a part of my https://github.com/cms-sw/cmssw/pull/21977#issuecomment-360923538 
``` define hpsPFTauIsolationMVArun2v1DBoldDMwLTTask in in HPSPFTaus_cff.py with 7 needed modules;
    add hpsPFTauDiscriminationByVVLooseIsolationMVArun2v1DBoldDMwLT only for this era```
although the text was probably not explicit enough to say that the latter action also should be applied in HPSPFTaus_cff.py

slava77 commented on RecoTauTag/Configuration/python/boostedHPSPFTaus_cfi.py:
```diff
@@ -21,7 +21,7 @@ def addBoostedTaus(process):
     process.ak4PFJetsRecoTauChargedHadronsBoosted.builders[1].dRconeLimitedToJetArea = cms.bool(True)
     process.combinatoricRecoTausBoosted.jetSrc = cms.InputTag('boostedTauSeeds')
-    process.combinatoricRecoTausBoosted.modifiers.remove(process.combinatoricRecoTausBoosted.modifiers[3])
+    #process.combinatoricRecoTausBoosted.modifiers.remove(process.combinatoricRecoTausBoosted.modifiers[3])
```
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant

BTW, this part was removed in the last version of the PR in the master.

roger-wolf committed:
adapting file to master -- removing of commented lines

roger-wolf committed:
removing final redundancies in module calls

cmsbuild commented:
Pull request #21977 was updated. @perrotta, @monttj, @cmsbuild, @franzoni, @slava77, @fabiocos, @davidlange6 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

perrotta commented:
Please solve the conflicts in PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py

cmsbuild commented:
-1

Tested at: d0977c8341aaad271cd311264f057148a322ba4e


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/3f87478f1c8d5a5b52e78212a7e35fed221d086d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26094/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26094/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26094/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
136.7611 step2
<pre>runTheMatrix-results/136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM/step2_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM.log

</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/3f87478f1c8d5a5b52e78212a7e35fed221d086d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26094/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26094/git-merge-result

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

roger-wolf committed:
resolving merge conflicts, caused by whitespace...

cmsbuild commented:
Pull request #21977 was updated. @perrotta, @monttj, @cmsbuild, @franzoni, @slava77, @fabiocos, @davidlange6 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: b5f1c48040411295d099958bdca1644dd499dd8b


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/b5f1c48040411295d099958bdca1644dd499dd8b
https://github.com/cms-sw/cmssw/commit/c73014f1cb638b15cca9fad6e476ed4d9f7a376a
https://github.com/cms-sw/cmssw/commit/ef3d4832597450cffc172c1bc825b891cac6176f
https://github.com/cms-sw/cmssw/commit/742139cd7695ef13b1638b6e2123c3b71b34f5bc
https://github.com/cms-sw/cmssw/commit/3f87478f1c8d5a5b52e78212a7e35fed221d086d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26106/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26106/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26106/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
136.7611 step2
<pre>runTheMatrix-results/136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM/step2_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM.log

</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/b5f1c48040411295d099958bdca1644dd499dd8b
https://github.com/cms-sw/cmssw/commit/c73014f1cb638b15cca9fad6e476ed4d9f7a376a
https://github.com/cms-sw/cmssw/commit/ef3d4832597450cffc172c1bc825b891cac6176f
https://github.com/cms-sw/cmssw/commit/742139cd7695ef13b1638b6e2123c3b71b34f5bc
https://github.com/cms-sw/cmssw/commit/3f87478f1c8d5a5b52e78212a7e35fed221d086d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26106/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26106/git-merge-result

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

perrotta commented:
Error in 136.7611 step2 looks like being related:
```
----- Begin Fatal Exception 15-Feb-2018 19:28:34 CET-----------------------
An exception of category 'InvalidReference' occurred while
   [0] Processing  Event run: 277069 lumi: 81 event: 36026102 stream: 0
   [1] Running path 'MINIAODoutput_step'
   [2] Prefetching for module PoolOutputModule/'MINIAODoutput'
   [3] Prefetching for module PATTauSlimmer/'slimmedTausBoosted'
   [4] Prefetching for module PATTauSelector/'selectedPatTausBoosted'
   [5] Calling method for module PATTauProducer/'patTausBoosted'
Exception Message:
AssociationVector: trying to use [] operator passing a reference
 with the wrong product id (i.e.: pointing to the wrong collection)
 keyRef.id = 3:1632, ref.id = 3:1631
----- End Fatal Exception -------------------------------------------------
```

roger-wolf commented:
Hi Andrea, 

I cannot reproduce the error on lxplus locally. I've copied the exact same config file (*) https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26106/runTheMatrix-results/136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM/step2_PAT_DQM.py and run on the exact same file, that is opened accorgin to the log files that prodiced the error. After 100 evt the run closes w/o any problems? @mbluj has also promised to look into the issue and might confirm my findings. 

Cheers,
Roger

mbluj committed:
Merged CMSSW_9_4_X_tau-pog_newTauID-MCv1 from repository cms-tau-pog with cms-merge-topic

mbluj committed:
move definition of the MVAIsoVVLoose tau-Id from miniAOD_tools.py to HPSPFTaus_cff.py

cmsbuild commented:
Pull request #21977 was updated. @perrotta, @monttj, @cmsbuild, @franzoni, @slava77, @fabiocos, @davidlange6 can you please check and sign again.

mbluj committed:
put the new 'run2_miniAOD_94XFall17' era in the same place as it is in master

cmsbuild commented:
Pull request #21977 was updated. @perrotta, @monttj, @cmsbuild, @franzoni, @slava77, @fabiocos, @davidlange6 can you please check and sign again.

mbluj commented:
Hello, 
the most recent commits should fix issue found by matrixTests and addresses a pending comment on order of definition of eras in `Configuration/StandardSequences/python/Eras.py`

perrotta commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 47d0d032e42877aeceb1f1dba07d394a58613b87
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26123/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/4554941087c69c6f3a9a62e0e3f32e496d2ba56e
https://github.com/cms-sw/cmssw/commit/b5f1c48040411295d099958bdca1644dd499dd8b
https://github.com/cms-sw/cmssw/commit/c73014f1cb638b15cca9fad6e476ed4d9f7a376a
https://github.com/cms-sw/cmssw/commit/ef3d4832597450cffc172c1bc825b891cac6176f
https://github.com/cms-sw/cmssw/commit/742139cd7695ef13b1638b6e2123c3b71b34f5bc
https://github.com/cms-sw/cmssw/commit/3f87478f1c8d5a5b52e78212a7e35fed221d086d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26123/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26123/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26123/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 260 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2718773
 - DQMHistoTests: Total failures: 135
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2718476
 - DQMHistoTests: Total skipped: 162
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0 KiB( 0 files compared)
 - Checked 107 log files, 9 edm output root files, 26 DQM output files

slava77 commented:
@smuzaffar 
comparisons link is pointing to a wrong place here as well

https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_9_4_MAOD_X_2018-02-15-1100+21977/25102/
which does not exist

instead of
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_9_4_X_2018-02-15-1100+21977/25102/
which exists

slava77 commented on RecoTauTag/Configuration/python/boostedHPSPFTaus_cfi.py:
```diff
@@ -21,12 +21,9 @@ def addBoostedTaus(process):
     process.ak4PFJetsRecoTauChargedHadronsBoosted.builders[1].dRcone = cms.double(0.3)
     process.ak4PFJetsRecoTauChargedHadronsBoosted.builders[1].dRconeLimitedToJetArea = cms.bool(True)
     process.combinatoricRecoTausBoosted.jetSrc = cms.InputTag('boostedTauSeeds')
-    process.combinatoricRecoTausBoosted.modifiers.remove(process.combinatoricRecoTausBoosted.modifiers[3])
```
removal of this line breaks compatibility for boostedTau reco for all eras.

I do not consider this change an obvious bugfix addressing something that was completely broken.

So, I propose to either reinstate this "modifiers.remove" or to apply it only to the new eras for which this PR is expected to be used in this 94X release (run2_miniAOD_80XLegacy, run2_miniAOD_94XFall17)

slava77 commented on RecoTauTag/Configuration/python/HPSPFTaus_cff.py:
```diff
@@ -848,7 +883,71 @@
+    )
+
+hpsPFTauDiscriminationByVVLooseIsolationMVArun2v1DBoldDMwLT = hpsPFTauDiscriminationByVLooseIsolationMVArun2v1DBoldDMwLT.clone()
+hpsPFTauDiscriminationByIsolationMVArun2v1DBoldDMwLTTaskExt = hpsPFTauDiscriminationByIsolationMVArun2v1DBoldDMwLTTask.copy()
```
please use a leading underscore here so that this module does not propagate downstream for ```import *```

```_hpsPFTauDiscriminationByIsolationMVArun2v1DBoldDMwLTTaskExt```

smuzaffar commented:
@slava77 , I have fixed the comparison job to upload the results in the correct area and restarted the comparison job.

mbluj commented:
Dear Slava,

On Fri, Feb 16, 2018 at 6:50 PM, Slava Krutelyov <notifications@github.com>
wrote:

> *@slava77* commented on this pull request.
> ------------------------------
>
> In RecoTauTag/Configuration/python/boostedHPSPFTaus_cfi.py
> <https://github.com/cms-sw/cmssw/pull/21977#discussion_r168825603>:
>
> > @@ -21,12 +21,9 @@ def addBoostedTaus(process):
>      process.ak4PFJetsRecoTauChargedHadronsBoosted.builders[1].dRcone = cms.double(0.3)
>      process.ak4PFJetsRecoTauChargedHadronsBoosted.builders[1].dRconeLimitedToJetArea = cms.bool(True)
>      process.combinatoricRecoTausBoosted.jetSrc = cms.InputTag('boostedTauSeeds')
> -    process.combinatoricRecoTausBoosted.modifiers.remove(process.combinatoricRecoTausBoosted.modifiers[3])
>
> removal of this line breaks compatibility for boostedTau reco for all eras.
>
> I do not consider this change an obvious bugfix addressing something that
> was completely broken.
>
BoostedTau reco is run only at MiniAOD step and is neither  part of reco
sequences nor RECO/AOD event content. Therefore, I think this fix does not
break any rules. However, if it does we can revert to previous state, i.e.
keep removing the modifier[3].

mbluj committed:
Add a leading underscore to name of an extended task

cmsbuild commented:
Pull request #21977 was updated. @perrotta, @monttj, @cmsbuild, @franzoni, @slava77, @fabiocos, @davidlange6 can you please check and sign again.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26123/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 260 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2718773
 - DQMHistoTests: Total failures: 135
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2718476
 - DQMHistoTests: Total skipped: 162
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0 KiB( 0 files compared)
 - Checked 107 log files, 9 edm output root files, 26 DQM output files

slava77 commented:
On 2/16/18 1:33 PM, mbluj wrote:
>> removal of this line breaks compatibility for boostedTau reco for all
> eras.
>>
>> I do not consider this change an obvious bugfix addressing something that
>> was completely broken.
>>
> BoostedTau reco is run only at MiniAOD step and is neither part of reco
> sequences nor RECO/AOD event content. Therefore, I think this fix does not
> break any rules. However, if it does we can revert to previous state, i.e.
> keep removing the modifier[3].
> 

the compatibility comment was with respect to miniAODv1.

The requirement here is less strict than for reco/aod,
 but it also looks like the modifier[3] feature is minor.

mbluj committed:
Restore removed modifier only for specific eras for backward compatibility

cmsbuild commented:
Pull request #21977 was updated. @perrotta, @monttj, @cmsbuild, @franzoni, @slava77, @fabiocos, @davidlange6 can you please check and sign again.

mbluj committed:
move modifers to correct location

Fix of previous commit: "restore removed modifier only for specific eras for backward compatibility".

cmsbuild commented:
Pull request #21977 was updated. @perrotta, @monttj, @cmsbuild, @franzoni, @slava77, @fabiocos, @davidlange6 can you please check and sign again.

mbluj commented:
Hello, I think that recent commits fix outstanding issues of the PR.

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: baffd683dd134233b655e82dc960ce6bbdb45c94
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26163/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/4554941087c69c6f3a9a62e0e3f32e496d2ba56e
https://github.com/cms-sw/cmssw/commit/b5f1c48040411295d099958bdca1644dd499dd8b
https://github.com/cms-sw/cmssw/commit/3f87478f1c8d5a5b52e78212a7e35fed221d086d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26163/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26163/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26163/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 222 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2718773
 - DQMHistoTests: Total failures: 157
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2718454
 - DQMHistoTests: Total skipped: 162
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0 KiB( 0 files compared)
 - Checked 107 log files, 9 edm output root files, 26 DQM output files

slava77 commented:
@smuzaffar 
based on the differences reported in this PR tests it looks like jenkins is using a baseline from CMSSW_9_4_X_2018-02-18-0000 instead of from CMSSW_9_4_MAOD_X_2018-02-18-0000.

Please check.
If that's the case, please let me know when the fix is available to pick up a correct baseline so that tests in this PR can be restarted.

smuzaffar commented:
please test
I have updated cms-bot to use MAOD ib for comparison

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: baffd683dd134233b655e82dc960ce6bbdb45c94
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26195/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/4554941087c69c6f3a9a62e0e3f32e496d2ba56e
https://github.com/cms-sw/cmssw/commit/b5f1c48040411295d099958bdca1644dd499dd8b
https://github.com/cms-sw/cmssw/commit/3f87478f1c8d5a5b52e78212a7e35fed221d086d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26195/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26195/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26195/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 117 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2718773
 - DQMHistoTests: Total failures: 103
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2718508
 - DQMHistoTests: Total skipped: 162
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0 KiB( 0 files compared)
 - Checked 107 log files, 9 edm output root files, 26 DQM output files

slava77 commented:
backport of #22105 restricted by modifiers, #21022 restricted by modifiers, and partly #21811

I'm missing the following:
- https://github.com/cms-sw/cmssw/pull/21022/files#diff-ec2ff1bf5cf3f3a19665b85a896dca25R121 change in PhysicsTools/PatAlgos/python/tools/tauTools.py in hpsTauIDSources list used in switchToPFTauHPS .. and eventually in AddTauCollection
    - do you need to support this part in 94X?

In retrospect, most of the updates could have been much easier if the payloads updates could have been handled via a global tag change in one place (instead of per-payload changes in RecoTauTag/Configuration/python/loadRecoTauTagMVAsFromPrepDB_cfi.py and everywhere they have to be consumed in the producers).

roger-wolf committed:
adding new working point for eras run2_miniAOD_80XLegacy and run2_miniAOD_94XFall17 to switchToPFTauHPS

cmsbuild commented:
Pull request #21977 was updated. @perrotta, @monttj, @cmsbuild, @franzoni, @slava77, @fabiocos, @davidlange6 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: e30e6ef9a09bad18cf8813ff9ba8cfd18c3ff308
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26211/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/4554941087c69c6f3a9a62e0e3f32e496d2ba56e
https://github.com/cms-sw/cmssw/commit/b5f1c48040411295d099958bdca1644dd499dd8b
https://github.com/cms-sw/cmssw/commit/3f87478f1c8d5a5b52e78212a7e35fed221d086d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26211/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26211/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21977/26211/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 117 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2718773
 - DQMHistoTests: Total failures: 107
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2718504
 - DQMHistoTests: Total skipped: 162
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0 KiB( 0 files compared)
 - Checked 107 log files, 9 edm output root files, 26 DQM output files

roger-wolf commented:
Dear colleagues, 

let me know whethere there is anything else pending to get this PR signed.

Cheers,
Roger

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/21977	e30e6ef
- backport of  #22105 restricted by modifiers, #21022 restricted by modifiers, and partly #21811 (the run2_miniAOD_94XFall17 modifier definition)
- jenkins tests pass and comparisons with baseline show differences only in the run2_miniAOD_80XLegacy-driven workflow 136.7611. The changes are in slimmedTausBoosted count/kinematics and IDs and in slimmedTaus IDs. The changes in the IDs are in the expected range of ID variables.

slava77 commented:
@fabiocos 
I wanted to check if this can be merged in 94MAODX. 
Please clarify.
Thank you.

fabiocos commented:
+operations

fabiocos commented:
+1

fabiocos commented:
merge

gpetruc commented:
Do I understand correctly that this PR *changes* the MVArun2v1DBoldDMwLT set of discriminators, rather than *adding* a new set?

slava77 commented:
> Do I understand correctly that this PR changes the MVArun2v1DBoldDMwLT set of discriminators, rather than adding a new set?

One VVL discriminator is added; the rest are indeed modified.

mbluj commented:
On Mon, Mar 19, 2018 at 2:23 PM, Slava Krutelyov <notifications@github.com>
wrote:

> Do I understand correctly that this PR changes the MVArun2v1DBoldDMwLT set
> of discriminators, rather than adding a new set?
>
> One VVL discriminator is added; the rest are indeed modified.
>
Yes, I confirm. It is what Tau POG wanted.

Michał

> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/21977#issuecomment-374209582>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AE-odTQs-snZMxJ2RFyPw18tqamiS8hVks5tf7E6gaJpZM4RulK1>
> .
>



-- 
Michał Bluj
National Centre for Nuclear Research / Narodowe Centrum Badań Jądrowych
Hoża 69
00-681 Warszawa, Poland
tel. (+48) 22 55 31 833
e-mail: michal.bluj@ncbj.gov.pl
